draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1536,229 +1485,97 @@ easy enough to parse with existing implementations (i.e., by splitting the file
-one another without loss of information. For example, a number of JSON-SEQ
-streamed qlogs could be combined into a JSON formatted qlog for later processing.
-Similarly, a captured binary qlog could be transformed to JSON for easier
-interpretation and sharing.
+Formal definition of additional qlog formats or encodings that use the
+optimization techniques described here, or any other optimization technique is
+left to future activity that can apply the following guidelines.
 
-Secondly, we can also consider other structured logging approaches that contain
-similar (though typically not identical) data to qlog, like raw packet capture
-files (for example .pcap files from tcpdump) or endpoint-specific logging formats
-(for example the NetLog format in Google Chrome). These are sometimes the only
-options, if an implementation cannot or will not support direct qlog output for
-any reason, but does provide other internal or external (e.g., SSLKEYLOGFILE
-export to allow decryption of packet captures) logging options For this second
-category, a (partial) transformation from/to qlog can also be defined.
+It is important that tooling can correctly parse and process serialized qlog. It
+is RECOMMENDED that new formats also define suitable file extensions and media types. This provides a clear signal and avoids the need to provide out-of-band information or to rely on heuristic fallbacks; see {{tooling}}.
 
-As such, when defining a new qlog serialization format or wanting to utilize
-qlog-compatible tools with existing codebases lacking qlog support, it is
-recommended to define and provide a concrete mapping from one format to default
-JSON-serialized qlog. Several of such mappings exist. Firstly,
-[pcap2qlog]((https://github.com/quiclog/pcap2qlog) transforms QUIC and HTTP/3
-packet capture files to qlog. Secondly,
-[netlog2qlog](https://github.com/quiclog/qvis/tree/master/visualizations/src/components/filemanager/netlogconverter)
-converts chromium's internal dictionary-encoded JSON format to qlog. Finally,
-[quictrace2qlog](https://github.com/quiclog/quictrace2qlog) converts the older
-quictrace format to JSON qlog. Tools can then easily integrate with these
-converters (either by incorporating them directly or for example using them as a
-(web-based) API) so users can provide different file types with ease. For example,
-the [qvis](https://qvis.edm.uhasselt.be) toolsuite supports a multitude of formats
-and qlog serializations.
+There is no established standard for indicating optimized formats. Therefore, it
+is RECOMMENDED that applied optimizations are expressed in ascending order of
+application. For example, if a qlog file using JSON is first optimized with
+technique A and then compressed with technique B, the resulting file would have
+the extension ".qlog.A.B". This allows tooling reverse the applied optimizations
+to arrive at the expected qlog representation. This recommendation applies to
+both file extensions and media types.
```

Comment:
I really wonder if we should keep this kind of thing in here, especially if we don't include any concrete examples anymore. Then we should also remove a similar paragraph up-top when discussing `qlog_format` for the first time.

Reply:
Happy to do that but let's track in a followup issue. Created #341